scene/3d/physics_body_3d.cpp

```diff
@@ -2925,6 +2937,9 @@ void PhysicalBone3D::_body_state_changed(PhysicsDirectBodyState3D *p_state) {
 
 	GDVIRTUAL_CALL(_integrate_forces, p_state);
 
+	linear_velocity = p_state->get_linear_velocity();
```

Comment:
I realize you copied the code from above the `_integrate_forces` call, but this looks odd to me. I don't see why this shouldn't synchronize as much mutable state as possible both before and after `_integrate_forces`, meaning velocities, `is_sleeping()` and `get_transform()`.

As it is right now, it looks like `PhysicalBone3D::_integrate_forces` doesn't actually get an up-to-date transform like it should be getting.

Reply:
I was also confused as to why the transform and sleeping were not synced here, should I sync them too?

Reply:
Also, PhysicalBone3D doesn't seem to have sleeping or inverse_inertia_tensor, we need to only sync transform or am I missing something?

Reply:
Cool, I'll do that